### PR TITLE
fix(core-magistrate-transactions): entity handler bootstrap method, fetch transactions using asset type/subType as number

### PR DESCRIPTION
**Pull request by air1one**
Entity handler bootstrap method was fetching transactions using asset type/subType as string, resulting in no transaction fetched at all.

Also refactored htlc handlers to remove htlc attribute when there is no more lock for the wallet.

Added state builder run() test to each functional test to check that it would rebuild the same wallets state that was built from running the functional tests.

### CODE REVIEW

File: __tests__/functional/transaction-forging/__support__/index.ts [hunk 1]
```diff
@@ -8,6 +9,7 @@ jest.setTimeout(1200000);
 
 import { DatabaseService } from "@arkecosystem/core-database";
 import { Sandbox } from "@packages/core-test-framework/src";
+import { StateBuilder } from "@arkecosystem/core-state/dist/state-builder";
```
Comment (rainydio) [inline __tests__/functional/transaction-forging/__support__/index.ts:12]: Bad reference

File: __tests__/functional/transaction-forging/__support__/index.ts [hunk 2]
```diff
@@ -92,10 +94,38 @@ export const setUp = async (): Promise<Contracts.Kernel.Application> => {
...
+            delete walletAttributes.delegate;
+        }
+        return {
+            publicKey: wallet.publicKey,
+            balance: wallet.balance,
+            nonce: wallet.nonce,
+            attributes: walletAttributes,
+        }
+    };
+    const sortWallets = (a: Contracts.State.Wallet, b: Contracts.State.Wallet) => a.publicKey! > b.publicKey! ? 1 : -1;
```
Comment (rainydio) [inline __tests__/functional/transaction-forging/__support__/index.ts:118]: `a.publicKey.localeCompare(b.publicKey)`

File: __tests__/unit/core-transactions/handlers/two/htlc-claim.test.ts [hunk 3]
```diff
@@ -392,8 +404,7 @@ describe("Htlc claim", () => {
 
                 await handler.apply(htlcClaimTransaction);
 
-                expect(lockWallet.getAttribute("htlc.locks")).toBeEmpty();
-                expect(lockWallet.getAttribute("htlc.lockedBalance")).toEqual(Utils.BigNumber.ZERO);
+                expect((lockWallet.getAttributes() as any).htlc).toBeUndefined();
```
Comment (rainydio) [inline __tests__/unit/core-transactions/handlers/two/htlc-claim.test.ts:407]: `expect(lockWallet.hasAttribute("htlc")).toBe(false);`

File: __tests__/functional/transaction-forging/__support__/index.ts (hunk 1 above)
Comment (air1one) [reply __tests__/functional/transaction-forging/__support__/index.ts]: :heavy_check_mark:

File: __tests__/functional/transaction-forging/__support__/index.ts (hunk 2 above)
Comment (air1one) [reply __tests__/functional/transaction-forging/__support__/index.ts]: :heavy_check_mark:

File: __tests__/unit/core-transactions/handlers/two/htlc-claim.test.ts (hunk 3 above)
Comment (air1one) [reply __tests__/unit/core-transactions/handlers/two/htlc-claim.test.ts]: :heavy_check_mark:

**Pull request merged**